vault/datadog_checks/vault/vault.py

```diff
@@ -0,0 +1,170 @@
+                    'tags': tags,
+                })
+            config['leader'] = current_leader
+
+    def check_health_v1(self, config, tags=None):
+        url = config['api_url'] + '/sys/health'
+        tags = tags or []
+        health_data = self.access_api(url, config, tags).json()
+
+        cluster_name = health_data.get('cluster_name')
+        if cluster_name:
+            tags.append('cluster_name:{}'.format(cluster_name))
+
+        vault_version = health_data.get('version')
+        if vault_version:
+            tags.append('vault_version:{}'.format(vault_version))
+
+        unsealed = not _is_affirmative(health_data.get('sealed'))
+        if unsealed:
+            self.service_check(self.SERVICE_CHECK_UNSEALED, AgentCheck.OK, tags=tags)
+        else:
+            self.service_check(self.SERVICE_CHECK_UNSEALED, AgentCheck.CRITICAL, tags=tags)
+
+        initialized = _is_affirmative(health_data.get('initialized'))
+        if initialized:
+            self.service_check(self.SERVICE_CHECK_INITIALIZED, AgentCheck.OK, tags=tags)
+        else:
+            self.service_check(self.SERVICE_CHECK_INITIALIZED, AgentCheck.CRITICAL, tags=tags)
+
+    def get_config(self, instance):
+        instance_id = hash_mutable(instance)
+        config = self.config.get(instance_id)
+        if config is None:
+            config = {}
+
+            try:
+                api_url = instance['api_url']
+                api_version = api_url[-1]
+                if api_version not in self.api_versions:
+                    self.log.warning(
+                        'Unknown Vault API version `{}`, using version '
+                        '`{}`'.format(api_version, self.DEFAULT_API_VERSION)
+                    )
+
+                config['api_url'] = api_url
+                config['api'] = self.api_versions.get(api_version, self.DEFAULT_API_VERSION)['functions']
+            except KeyError:
+                self.log.error('Vault configuration setting `api_url` is required')
+                return
+
+            client_token = instance.get('client_token')
+            config['headers'] = {'X-Vault-Token': client_token} if client_token else None
+
+            username = instance.get('username')
+            password = instance.get('password')
+            config['auth'] = (username, password) if username and password else None
+
+            config['ssl_verify'] = instance.get('ssl_verify', True)
+            config['proxies'] = self.get_instance_proxy(instance, config['api_url'])
+            config['timeout'] = int(instance.get('timeout', 20))
+            config['tags'] = instance.get('tags', [])
+
+            # Keep track of the previous cluster leader to detect changes.
+            config['leader'] = None
+            config['detect_leader'] = _is_affirmative(instance.get('detect_leader'))
+
+            self.config[instance_id] = config
+
+        return config
+
+    def access_api(self, url, config, tags=None):
```

Comment:
Is it possible for ``tags`` to equal ``None``? Not opposed to leaving it like this, however.

Reply:
It is impossible, but yes I'll just make it required.